tests/testthat/test_simulate_data.R

```diff
@@ -0,0 +1,75 @@
+test_that("simulated data", {
+    # So far we are skipping tests on these platforms until
+    # we find an efficient way to update rstan testthat snapshots on all of them
+    skip_on_os(c("windows", "mac"))
+    skip_on_ci()
+
+    library(dplyr)
+    library(serofoi)
+
+    seed <- 1234
+    size_age_class <- 5
+    #----- Test for constant FoI
+    # foi_model <- "constant"
+    # foi_sim <- rep(0.02, 50)
+    # case_label <- "constant_foi_"
+    # max_lambda <- 0.035
+
+    #----- Test for stepwise decreasing FoI
+    foi_model = "tv_normal"
+    no_transm <- 0.0000000001
+    foi_sim <- c(rep(0.2, 25), rep(0.1, 10), rep(no_transm, 15))
+    case_label <- "sw_dec_foi_"
+    max_lambda <- 0.3
+
+    #----- Results paths
+    path_data_no_grouped <- testthat::test_path(
+      "extdata", paste0(case_label, "sim_data_no_grouped.csv")
+    )
+
+    path_data_grouped <- testthat::test_path(
+      "extdata", paste0(case_label, "sim_data_grouped.csv")
+    )
+
+    #----- Test function generate_sim_data
+    sim_data <- generate_sim_data(foi = foi_sim,
+                                  size_age_class = size_age_class,
+                                  tsur = 2050,
+                                  birth_year_min = 2000,
+                                  survey_label = 'foi_sim',
+                                  seed = seed)
```

Comment:
I don't see the checks on this generated data? E.g. its dimensions, its values and so on.

Reply:
We also need tests of the individual functions which `generate_sim_data` calls. (That's a key reason of splitting this function up into subfunctions -- we can test them rather than just the aggregate output.)